core/ledger/kvledger/txmgmt/statedb/stateleveldb/stateleveldb.go

```diff
@@ -220,6 +221,41 @@ func (vdb *versionedDB) GetFullScanIterator(skipNamespace func(string) bool) (st
 	return newFullDBScanner(vdb.db, skipNamespace)
 }
 
+// ImportState implements method in VersionedDB interface. The function is expected to be used
+// for importing the state from a previously snapshotted state. The parameter itr provides access to
+// the snapshotted state.
+func (vdb *versionedDB) ImportState(itr statedb.FullScanIterator, dbValueFormat byte) error {
+	if dbValueFormat != fullScanIteratorValueFormat {
+		return errors.Errorf("value format [%x] not supported. Expected value format [%x]",
+			dbValueFormat, fullScanIteratorValueFormat)
+	}
+	dbBatch := vdb.db.NewUpdateBatch()
+	batchSize := 0
+	for {
+		compositeKey, dbValue, err := itr.Next()
+		if err != nil {
+			return err
+		}
+		if compositeKey == nil {
+			break
+		}
+		dataKey := encodeDataKey(compositeKey.Namespace, compositeKey.Key)
+		batchSize += len(dataKey) + len(dbValue)
+		dbBatch.Put(dataKey, dbValue)
+		if batchSize >= maxDataImportBatchSize {
+			if err := vdb.db.WriteBatch(dbBatch, true); err != nil {
+				return err
+			}
```

Comment:
missed to reset the batchSize to 0.

Reply:
yes, In fact I was thinking of moving this size field inside batch so we could call `batch.Size` and `batch.Reset` but eventually missed at both the places :-)